Tests/GRPCTests/ServerErrorDelegateTests.swift

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import XCTest
+import NIO
+import NIOHTTP1
+import GRPC
+import EchoModel
+import EchoImplementation
+import Logging
+
+private class ServerErrorDelegateMock: ServerErrorDelegate {
+  var transformLibraryErrorCallCount: Int = 0
+  var transformLibraryErrorHandler: ((Error) -> (GRPCStatusAndMetadata?))?
+  func transformLibraryError(_ error: Error) -> GRPCStatusAndMetadata? {
+    transformLibraryErrorCallCount += 1
+    if let transformLibraryErrorHandler = transformLibraryErrorHandler {
+        return transformLibraryErrorHandler(error)
+    }
+    fatalError("transformLibraryError returns can't have a default value thus its handler must be set")
+  }
+}
+
+class ServerErrorDelegateTests: GRPCTestCase {
+  private var channel: EmbeddedChannel!
+  private var errorDelegate: ServerErrorDelegateMock!
```

Comment:
Rather than holding on to the error delegate like this, defining a `ServerErrorDelegate` which accepts a closure on `init` would be neater; you can completely get rid of the `fatalError` and optional check here. The setup would then be done in a setup function called from the test case.

Reply:
I think we have to hold the error delegate one way or another because it is defined as a `weak var errorDelegate: ServerErrorDelegate?` inside `GRPCServerRequestRoutingHandler`. If we are not holding it in the test case it will be immediately deallocated.